packages/fxa-settings/src/pages/SigninTotpCode/index.tsx

```diff
@@ -0,0 +1,152 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React, { useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { RouteComponentProps } from '@reach/router';
+import InputText from '../../components/InputText';
+import { FtlMsg } from 'fxa-react/lib/utils';
+import { useFtlMsgResolver } from '../../models/hooks';
+import { usePageViewEvent, logViewEvent } from '../../lib/metrics';
+// import { useAlertBar } from '../../models';
+import { ReactComponent as TwoFactorImg } from './graphic_two_factor_auth.svg';
+import CardHeader from '../../components/CardHeader';
+
+// --serviceName-- is the relying party
+
+export type SigninTotpCodeProps = { serviceName?: string };
+
+type FormData = {
+  confirmationCode: string;
+};
+
+const SigninTotpCode = ({
+  serviceName,
+}: SigninTotpCodeProps & RouteComponentProps) => {
+  usePageViewEvent('signin-totp-code', {
+    entrypoint_variation: 'react',
+  });
+
+  const [securityCode, setSecurityCode] = useState<string>('');
+  const [totpErrorMessage, setTotpErrorMessage] = useState<string>('');
+  const [isFocused, setIsFocused] = useState(false);
+  // const alertBar = useAlertBar();
+  const ftlMsgResolver = useFtlMsgResolver();
+
+  const onFocusMetricsEvent = 'signin-totp-code.engage';
+
+  const onFocus = () => {
+    if (!isFocused && onFocusMetricsEvent) {
+      logViewEvent('flow', onFocusMetricsEvent, {
+        entrypoint_variation: 'react',
+      });
+      setIsFocused(true);
+    }
+  };
+
+  const { handleSubmit } = useForm<FormData>({
+    mode: 'onBlur',
+    criteriaMode: 'all',
+    defaultValues: {
+      confirmationCode: '',
+    },
+  });
+
+  const onSubmit = () => {
+    if (!securityCode) {
+      const codeRequiredError = ftlMsgResolver.getMsg(
+        'signin-totp-code-required-error',
+        'Two-step authentication code required'
+      );
+      setTotpErrorMessage(codeRequiredError);
+    }
+    try {
+      // Check security code
+      // Log success event
+      // Check if isForcePasswordChange
+    } catch (e) {
+      // TODO: error handling, error message confirmation
+      // const errorSigninTotpCode = ftlMsgResolver.getMsg(
+      //   'signin-totp-code-error-general',
+      //   'Invalid confirmation code'
+      // );
+      // alertBar.error(errorSigninTotpCode);
```

Comment:
I wonder if we'll want to use the alert bar on these pages vs the error div that usually sits within the cards themselves, hmm.

Reply:
Could be up for future decision, when we hook up functionality and test?